zcash_primitives/src/merkle_tree.rs

```diff
@@ -25,6 +29,47 @@ pub trait Hashable: Clone + Copy {
     fn empty_root(_: usize) -> Self;
 }
 
+/// A hashable node within a Merkle tree.
+pub trait HashSer {
+    /// Parses a node from the given byte source.
+    fn read<R: Read>(reader: R) -> io::Result<Self>
+    where
+        Self: Sized;
+
+    /// Serializes this node.
+    fn write<W: Write>(&self, writer: W) -> io::Result<()>;
+}
+
+impl<T> Hashable for T
+where
+    T: incrementalmerkletree::Hashable + HashSer + Copy,
+{
+    /// Parses a node from the given byte source.
+    fn read<R: Read>(reader: R) -> io::Result<Self> {
+        <Self as HashSer>::read(reader)
+    }
+
+    /// Serializes this node.
+    fn write<W: Write>(&self, writer: W) -> io::Result<()> {
+        <Self as HashSer>::write(self, writer)
+    }
+
+    /// Returns the parent node within the tree of the two given nodes.
+    fn combine(alt: usize, lhs: &Self, rhs: &Self) -> Self {
+        <Self as incrementalmerkletree::Hashable>::combine(Altitude::from(alt as u8), lhs, rhs)
```

Comment:
This should never be a concern (and we'll probably remove this trait soon anyway), but just for correctness:
```suggestion
        <Self as incrementalmerkletree::Hashable>::combine(Altitude::from(u8::try_from(alt).unwrap()), lhs, rhs)
```